File: frameworks-orm/express-orm/src/entities/Post.ts
import { Entity, PrimaryColumn, Column, OneToMany, ManyToOne } from "typeorm";

@Entity("tb_posts")
export class Post {
    @PrimaryColumn()
    id: string;

    @Column()
    title: string;

    @Column({ type: "text" })
    content: string;

    @Column({ name: "author_id" })
    authorId: string;

    // OPTIMIZED: Use eager loading to prevent N+1 queries
    @ManyToOne(() => User, user => user.posts, {
        eager: true  // Preload author relationship
    })
    author: User;

    // OPTIMIZED: Use eager loading for comments
    @OneToMany(() => Comment, comment => comment.post, {
        eager: true  // Preload comments relationship
    })
    comments: Comment[];
}

File: frameworks/gin-rest/internal/models/post.go
package models

// Post represents a post in the system
type Post struct {
	ID       string    `json:"id" db:"id"`
	Title    string    `json:"title" db:"title"`
	Content  *string   `json:"content,omitempty" db:"content"`
	AuthorID string    `json:"author_id" db:"author_id"`
	Author   *User     `json:"author,omitempty"`
	Comments []Comment `json:"comments,omitempty"`
}

// PostResponse represents the post data returned in API responses
type PostResponse struct {
	ID       string    `json:"id"`
	Title    string    `json:"title"`
	Content  *string   `json:"content,omitempty"`
	Author   *User     `json:"author,omitempty"`
	Comments []Comment `json:"comments,omitempty"`
}
